activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -316,4 +316,4 @@ def set_defaults(target) # :doc:
   end
 end
 
-HashWithIndifferentAccess = ActiveSupport::HashWithIndifferentAccess
+HashWithIndifferentAccess = ActiveSupport::Deprecation::DeprecatedConstantProxy.new("HashWithIndifferentAccess", "ActiveSupport::HashWithIndifferentAccess")
```

Comment:
This will broke the ability to inherit from this constant. Can you add a test for it and fix?